src/Microsoft.Azure.SignalR.Management/Serialization/JsonHubProtocolResolver.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+#nullable enable
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    /// <summary>
+    /// Modifed based on https://github.com/dotnet/aspnetcore/blob/d9660d157627af710b71c636fa8cb139616cadba/src/SignalR/server/Core/src/Internal/DefaultHubProtocolResolver.cs
+    /// Compared to the origin protocol resolver, it adds logic to decides whether to replace <see cref="JsonHubProtocol"/> with <see cref="JsonObjectSerializerHubProtocol"/>.
+    /// </summary>
+    internal class JsonHubProtocolResolver : IHubProtocolResolver
+    {
+        private readonly ILogger<JsonHubProtocolResolver> _logger;
+        private readonly List<IHubProtocol> _hubProtocols;
+        private readonly Dictionary<string, IHubProtocol> _availableProtocols;
+
+        public IReadOnlyList<IHubProtocol> AllProtocols => _hubProtocols;
+
+        public JsonHubProtocolResolver(IEnumerable<IHubProtocol> availableProtocols, ILogger<JsonHubProtocolResolver> logger)
+        {
+            _logger = logger ?? NullLogger<JsonHubProtocolResolver>.Instance;
+            _availableProtocols = new Dictionary<string, IHubProtocol>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var protocol in availableProtocols)
+            {
+                // decide whether replace the default JSON protocol.
+                if (protocol is JsonObjectSerializerHubProtocol serializerHubProtocol && serializerHubProtocol.Usable is not true)
+                {
+                    Log.SkipJsonObjectSerializerProtocol(_logger);
+                    continue;
+                }
+                _availableProtocols[protocol.Name] = protocol;
+                Log.RegisteredSignalRProtocol(_logger, protocol.Name, protocol.GetType());
```

Comment:
what if protocol name duplicates, the latter one overrides the previous one but the log is misleading